File: trunk/source/add-lib-develop/add-lib-develop/core/src/test/resources/TestProgram.cs
using System;

namespace TestCSharp {

    class MainClass {
        public static void Main(string[] args) {
            Predicates test = new PredicatesABC();
            if (SumOfPredicatesABC.sumOfPredicatesABC(test).Equals("2.0")) {
                System.Environment.Exit(0);
            } else {
                System.Environment.Exit(1);
            }
        }
    }

    class PredicatesABC : Predicates {
      public bool predA() {
        return true;
      }

      public bool predB() {
        return false;
      }

      public bool predC() {
        return true;
      }
    }

}

File: trunk/source/add-lib-develop/add-lib-develop/core/src/test/resources/TestProgram.cpp
#include "PredicatesPlusPlus.h"
#include "SumOfPredicatesABC.h"

bool predA() {
	return true;
}

bool predB() {
	return false;
}

bool predC() {
	return true;
}

int main() {
	Predicates test(&predA, &predB, &predC);
	SumOfPredicatesABC abc;
	if (abc.sumOfPredicatesABC(test) == "2.0")
		return 0;
  else
		return 1;
}
